scripts/missing-ingest-data-fixer/ingest_query.py

```diff
@@ -0,0 +1,180 @@
+                new_dict[headers[i]] = row[i]
+            list_projects[new_dict.get('UUID')] = new_dict
+
+        return list_projects
+
+    @staticmethod
+    def not_exists_by_doi(ingest_projects, dts_projects):
+        exists = False
+        not_in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                for publication in project_ingest.get('content', {}).get('publications', []):
+                    if str(publication.get('doi')).lower() == str(project_dts.get('doi')).lower():
+                        exists = True
+                        break
+                if exists:
+                    exists = False
+                    break
+            else:
+                not_in_ingest[project_dts['dcp_id']] = project_dts
+
+        return not_in_ingest
+
+    @staticmethod
+    def exists_by_doi(ingest_projects, dts_projects):
+        exists = False
+        in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                for publication in project_ingest.get('content', {}).get('publications', []):
+                    if str(publication.get('doi')).lower() == str(project_dts.get('doi')).lower():
+                        exists = True
+                        in_ingest[project_ingest.get('uuid').get('uuid')] = project_dts
+                        break
+                if exists:
+                    exists = False
+                    break
+
+        return in_ingest
+
+    @staticmethod
+    def not_exists_by_geo_accession(ingest_projects, dts_projects):
+        exists = False
+        not_in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                for geo in project_ingest.get('content', {}).get('geo_series_accessions', []):
+                    if geo in project_dts['data_accession']:
+                        exists = True
+                        break
+                if exists:
+                    exists = False
+                    break
+            else:
+                not_in_ingest[project_dts['dcp_id']] = project_dts
+
+        return not_in_ingest
+
+    @staticmethod
+    def exists_by_geo_accession(ingest_projects, dts_projects):
+        exists = False
+        in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
+                for geo in project_ingest.get('content', {}).get('geo_series_accessions', []):
+                    if geo in project_dts['data_accession']:
+                        exists = True
+                        in_ingest[project_ingest.get('uuid').get('uuid')] = project_dts
+                        break
+                if exists:
+                    exists = False
+                    break
+
+        return in_ingest
+
+    @staticmethod
+    def filter_by_is_in_catalogue(ingest_projects, dts_projects):
+        not_in_cat = []
```

Comment:
could have just use the whole word for catalogue for clarity

Reply:
Also, I am not using this for organ fix.